test/functional/normal/fold_spec.lua

```diff
@@ -44,4 +47,114 @@ describe('Folds', function()
     8
     9]])
   end)
+  it('updates correctly on :read', function()
+    -- luacheck: ignore
+    helpers.write_file('./to_read.txt', [[
+    a
+
+
+    	a]])
+    -- luacheck: ignore
+    insert([[
+    	a
+    	a
+    	a
+    	a
+    ]])
+    execute('set foldmethod=indent', '2', '%foldopen')
+    execute('read ./to_read.txt')
+    -- Just to check we have the correct file text.
+    -- luacheck: ignore
+    expect([[
+    	a
+    	a
+    a
+
+
+    	a
+    	a
+    	a
+    ]])
+    for i = 1,2 do
+      eq(1, funcs.foldlevel(i))
+    end
+    for i = 3,5 do
+      eq(0, funcs.foldlevel(i))
+    end
+    for i = 6,8 do
+      eq(1, funcs.foldlevel(i))
+    end
+    os.remove('to_read.txt')
```

Comment:
Should be in `after_each` in case the test fails. If the file doesn't exist it won't complain.